### ISSUE TITLE
`another module with an equal name` warning - need more info

### ISSUE THREAD

**Issue by wclr** [webpack/webpack#2362]
There is warning sometimes:

```
(webpack)/buildin/module.js
There is another module with an equal name when case is ignored.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Rename module if multiple modules are expected or use equal casing if one module is expected.

/Code/app/~/webpack/buildin/module.js
There is another module with an equal name when case is ignored.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Rename module if multiple modules are expected or use equal casing if one module is expected.
```

it would useful to have additional info what does this warning mean and what causes it.
In my case this is caused by linked packages that require module with the same name from different locations. But it would be nice to have this **conflicting module name** in the warning message and probably some additional some details.

**Comment by drcmda** [comment]
Same issue. I have no idea why or where it came from.

**Comment by whitecolor** [comment]
do you use some npm linked modules?

**Comment by thorn0** [comment]
@whitecolor I got this warning with the exactly same module names: `(webpack)/buildin/module.js` and `~/webpack/buildin/module.js`. Which modules caused it in your case?

**Comment by thorn0** [comment]
It turns out, this happens only if I run `webpack-web-server` via `npm run ...`.
If I run it as `node_modules\.bin\webpack-dev-server`, everything is OK.

**Comment by whitecolor** [comment]
that is strange =) btw I don't use `webpack-dev-server` cli because problems with hot reload I meat https://github.com/webpack/webpack/issues/2424

**Comment by thorn0** [comment]
@whitecolor What's your OS?

**Comment by whitecolor** [comment]
(Windows 10 x64, Node v5.5.0)

**Comment by thorn0** [comment]
Something tells me these issues are Windows-specific.

**Comment by whitecolor** [comment]
I'm not sure, but might be.

**Comment by sokra** [comment]
The warning is windows and mac. In linux the same combination would cause a module not found error. Linux users are usually more aware for file system casing...

**Comment by whitecolor** [comment]
@sokra If i'm correct I had this problem with linked modules (modules with the same name where take from different locations on the disk) could it be the issue?

**Comment by thorn0** [comment]
@sokra I understand the original meaning of this error message. The thing is that it's caused by something else in this case, not by a file name casing issue. As I wrote, it happens only if I run webpack-web-server via `npm run`. So I'm almost sure it's some kind of Windows-specific path conversion issue.

**Comment by thorn0** [comment]
Sorry, I forgot that when I run the dev server via `npm run`, I run it with `--inline`. So that's the difference, that's what leads to this warning: the inline mode.

**Comment by thorn0** [comment]
So, I've isolated the issue.

package.json:
```js
{
  "devDependencies": { "webpack": "1.13.0", "webpack-dev-server": "1.14.1" },
  "dependencies": { "es6-promise": "3.1.2" }
} 
```

webpack.config.js:
```js
module.exports = {
    entry: './app.js',
    output: { path: __dirname, filename: 'bundle.js', pathinfo: true }
};
```

app.js:
```js
require('es6-promise');
```

Now, if I run `node_modules\.bin\webpack-dev-server --inline`, here is what I get. `buildin/module.js` is indeed included twice: as [4] and as [76].
```
http://localhost:8080/
webpack result is served from /
content is served from d:\dev\foo
Hash: bde29dc039a20f590eab
Version: webpack 1.13.0
Time: 868ms
    Asset    Size  Chunks             Chunk Names
bundle.js  285 kB       0  [emitted]  main
chunk    {0} bundle.js (main) 246 kB [rendered]
    [0] multi main 40 bytes {0} [built]
    [1] (webpack)-dev-server/client?http://localhost:8080 2.67 kB {0} [built]
    [2] D:/dev/foo/~/url/url.js 22.3 kB {0} [built]
    [3] D:/dev/foo/~/url/~/punycode/punycode.js 14.6 kB {0} [built]
    [4] (webpack)/buildin/module.js 251 bytes {0} [built]
    [5] D:/dev/foo/~/querystring/index.js 127 bytes {0} [built]
    [6] D:/dev/foo/~/querystring/decode.js 2.4 kB {0} [built]
    [7] D:/dev/foo/~/querystring/encode.js 2.09 kB {0} [built]
    [8] D:/dev/foo/~/sockjs-client/lib/entry.js 244 bytes {0} [built]
    [9] D:/dev/foo/~/sockjs-client/lib/transport-list.js 613 bytes {0} [built]
   [10] D:/dev/foo/~/sockjs-client/lib/transport/websocket.js 2.72 kB {0} [built]
   [11] D:/dev/foo/~/process/browser.js 2.06 kB {0} [built]
   [12] D:/dev/foo/~/sockjs-client/lib/utils/event.js 2 kB {0} [built]
   [13] D:/dev/foo/~/sockjs-client/lib/utils/random.js 746 bytes {0} [built]
   [14] D:/dev/foo/~/sockjs-client/lib/utils/browser-crypto.js 438 bytes {0} [built]
   [15] D:/dev/foo/~/sockjs-client/lib/utils/url.js 975 bytes {0} [built]
   [16] D:/dev/foo/~/url-parse/index.js 8.09 kB {0} [built]
   [17] D:/dev/foo/~/requires-port/index.js 753 bytes {0} [built]
   [18] D:/dev/foo/~/url-parse/lolcation.js 1.58 kB {0} [built]
   [19] D:/dev/foo/~/querystringify/index.js 1.3 kB {0} [built]
   [20] D:/dev/foo/~/debug/browser.js 3.76 kB {0} [built]
   [21] D:/dev/foo/~/debug/debug.js 4.1 kB {0} [built]
   [22] D:/dev/foo/~/ms/index.js 2.33 kB {0} [built]
   [23] D:/dev/foo/~/inherits/inherits_browser.js 672 bytes {0} [built]
   [24] D:/dev/foo/~/sockjs-client/lib/event/emitter.js 1.27 kB {0} [built]
   [25] D:/dev/foo/~/sockjs-client/lib/event/eventtarget.js 1.85 kB {0} [built]
   [26] D:/dev/foo/~/sockjs-client/lib/transport/browser/websocket.js 58 bytes {0} [built]
   [27] D:/dev/foo/~/sockjs-client/lib/transport/xhr-streaming.js 1.25 kB {0} [built]
   [28] D:/dev/foo/~/sockjs-client/lib/transport/lib/ajax-based.js 1.31 kB {0} [built]
   [29] D:/dev/foo/~/sockjs-client/lib/transport/lib/sender-receiver.js 1.15 kB {0} [built]
   [30] D:/dev/foo/~/sockjs-client/lib/transport/lib/buffered-sender.js 2.3 kB {0} [built]
   [31] D:/dev/foo/~/sockjs-client/lib/transport/lib/polling.js 1.32 kB {0} [built]
   [32] D:/dev/foo/~/sockjs-client/lib/transport/receiver/xhr.js 1.58 kB {0} [built]
   [33] D:/dev/foo/~/sockjs-client/lib/transport/sender/xhr-cors.js 343 bytes {0} [built]
   [34] D:/dev/foo/~/sockjs-client/lib/transport/browser/abstract-xhr.js 4.8 kB {0} [built]
   [35] D:/dev/foo/~/sockjs-client/lib/transport/sender/xhr-local.js 352 bytes {0} [built]
   [36] D:/dev/foo/~/sockjs-client/lib/utils/browser.js 560 bytes {0} [built]
   [37] D:/dev/foo/~/sockjs-client/lib/transport/xdr-streaming.js 984 bytes {0} [built]
   [38] D:/dev/foo/~/sockjs-client/lib/transport/sender/xdr.js 2.46 kB {0} [built]
   [39] D:/dev/foo/~/sockjs-client/lib/transport/eventsource.js 766 bytes {0} [built]
   [40] D:/dev/foo/~/sockjs-client/lib/transport/receiver/eventsource.js 1.58 kB {0} [built]
   [41] D:/dev/foo/~/sockjs-client/lib/transport/browser/eventsource.js 37 bytes {0} [built]
   [42] D:/dev/foo/~/sockjs-client/lib/transport/lib/iframe-wrap.js 981 bytes {0} [built]
   [43] D:/dev/foo/~/sockjs-client/lib/transport/iframe.js 3.83 kB {0} [built]
   [44] D:/dev/foo/~/json3/lib/json3.js 43.3 kB {0} [built]
   [45] (webpack)/buildin/amd-options.js 43 bytes {0} [built]
   [46] D:/dev/foo/~/sockjs-client/lib/version.js 26 bytes {0} [built]
   [47] D:/dev/foo/~/sockjs-client/lib/utils/iframe.js 5.06 kB {0} [built]
   [48] D:/dev/foo/~/sockjs-client/lib/utils/object.js 532 bytes {0} [built]
   [49] D:/dev/foo/~/sockjs-client/lib/transport/htmlfile.js 710 bytes {0} [built]
   [50] D:/dev/foo/~/sockjs-client/lib/transport/receiver/htmlfile.js 2.2 kB {0} [built]
   [51] D:/dev/foo/~/sockjs-client/lib/transport/xhr-polling.js 894 bytes {0} [built]
   [52] D:/dev/foo/~/sockjs-client/lib/transport/xdr-polling.js 712 bytes {0} [built]
   [53] D:/dev/foo/~/sockjs-client/lib/transport/jsonp-polling.js 1.02 kB {0} [built]
   [54] D:/dev/foo/~/sockjs-client/lib/transport/receiver/jsonp.js 5.57 kB {0} [built]
   [55] D:/dev/foo/~/sockjs-client/lib/transport/sender/jsonp.js 2.46 kB {0} [built]
   [56] D:/dev/foo/~/sockjs-client/lib/main.js 11.9 kB {0} [built]
   [57] D:/dev/foo/~/sockjs-client/lib/shims.js 18.2 kB {0} [built]
   [58] D:/dev/foo/~/sockjs-client/lib/utils/escape.js 2.31 kB {0} [built]
   [59] D:/dev/foo/~/sockjs-client/lib/utils/transport.js 1.35 kB {0} [built]
   [60] D:/dev/foo/~/sockjs-client/lib/utils/log.js 450 bytes {0} [built]
   [61] D:/dev/foo/~/sockjs-client/lib/event/event.js 477 bytes {0} [built]
   [62] D:/dev/foo/~/sockjs-client/lib/location.js 177 bytes {0} [built]
   [63] D:/dev/foo/~/sockjs-client/lib/event/close.js 295 bytes {0} [built]
   [64] D:/dev/foo/~/sockjs-client/lib/event/trans-message.js 292 bytes {0} [built]
   [65] D:/dev/foo/~/sockjs-client/lib/info-receiver.js 2.22 kB {0} [built]
   [66] D:/dev/foo/~/sockjs-client/lib/transport/sender/xhr-fake.js 456 bytes {0} [built]
   [67] D:/dev/foo/~/sockjs-client/lib/info-iframe.js 1.52 kB {0} [built]
   [68] D:/dev/foo/~/sockjs-client/lib/info-iframe-receiver.js 791 bytes {0} [built]
   [69] D:/dev/foo/~/sockjs-client/lib/info-ajax.js 1.03 kB {0} [built]
   [70] D:/dev/foo/~/sockjs-client/lib/iframe-bootstrap.js 2.9 kB {0} [built]
   [71] D:/dev/foo/~/sockjs-client/lib/facade.js 723 bytes {0} [built]
   [72] D:/dev/foo/~/strip-ansi/index.js 161 bytes {0} [built]
   [73] D:/dev/foo/~/ansi-regex/index.js 135 bytes {0} [built]
   [74] ./app.js 25 bytes {0} [built]
   [75] ./~/es6-promise/dist/es6-promise.js 32.2 kB {0} [built]
   [76] ./~/webpack/buildin/module.js 251 bytes {0} [built]
   [78] (webpack)/buildin/amd-define.js 85 bytes {0} [built]
     + 1 hidden modules

WARNING in ./~/webpack/buildin/module.js
There is another module with an equal name when case is ignored.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Rename module if multiple modules are expected or use equal casing if one module is expected.

WARNING in (webpack)/buildin/module.js
There is another module with an equal name when case is ignored.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Rename module if multiple modules are expected or use equal casing if one module is expected.
webpack: bundle is now VALID.
```

**Comment by thorn0** [comment]
Tested this configuration on Mac OS X 10.9.5. The issue doesn't exist there. `module.js` is included only once.

**Comment by sokra** [comment]
`d:\dev\foo` vs `D:/dev/foo`

not sure why...

**Comment by thorn0** [comment]
So it's actually caused by the fact that Node's builtin functions aren't consistent in what case they use for the drive letters on Windows. `__filename`, `__dirname`, and `require.resolve` return it in upper case (`D:\dev\foo`) whereas `path.resolve` and `process.cwd` in lower (`d:\dev\foo`).

webpack-dev-server uses `require.resolve` [here](https://github.com/webpack/webpack-dev-server/blob/35286d8c5cd6a891c070cb9532ca944744d330d9/bin/webpack-dev-server.js#L232), which leads to the bug I described above.

I think the best workaround here would be to make webpack internally consider `D:\file.js` and `d:\file.js` exactly same files.

**Comment by sokra** [comment]
For me, `process.cwd()` returns `"C:\\Users\\sokra"` in with an uppercase letter and correct slashes. How do you call node? (from cmd?) What's the result of `echo %cd%`?

**Comment by thorn0** [comment]
For me, it returns lower case only for drive D. All the other drive letters are uppercase. And it happens not only in Node. It's something on the OS level:
```
C:\>echo %cd%
C:\

C:\>D:

d:\dev>echo %cd%
d:\dev

d:\dev>
```

**Comment by thorn0** [comment]
Yes, I called Node from cmd. And in the properties of the shortcut for cmd.exe, the "Start in" folder was specified with a lowercase "d".

**Comment by thorn0** [comment]
test.js:
```js
var exprs = ['__dirname', '__filename', 'require.resolve("./test")', 
    'require("path").resolve("test.js")', 'process.cwd()'];
for (var expr of exprs) {
    console.log(expr, eval(expr));
}
```

In cmd:
```
d:\dev\foo>node test
__dirname D:\dev\foo
__filename D:\dev\foo\test.js
require.resolve("./test") D:\dev\foo\test.js
require("path").resolve("test.js") d:\dev\foo\test.js
process.cwd() d:\dev\foo
```

If I change the case of the drive letter in the "Start in" field of the shortcut for cmd, everything becomes consistent.
```
D:\dev\foo>node test
__dirname D:\dev\foo
__filename D:\dev\foo\test.js
require.resolve("./test") D:\dev\foo\test.js
require("path").resolve("test.js") D:\dev\foo\test.js
process.cwd() D:\dev\foo
```

**Comment by sokra** [comment]
hmm weird...

I've checked how node.js behaves, and it also returns two modules for different drive letters. So webpacks behavior is equal to node.js.

So this should be discussed/fixed on node.js level.

**Comment by drcmda** [comment]
@thorn0 This was it, thanks! It's the weirdest thing, though. I usually type cd **c:\**dev\projectname into my console and get going. Typing cd **C:\**dev\projectname fixed it. I didn't even know it was possible, or made a difference.

**Comment by whitecolor** [comment]
I've got a like warnings for style loader (actually loading less)

> WARNING in ./client/~/style-loader/addStyles.js
There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* D:\Code\app\client\node_modules\style-loader\addStyles.js
* d:\Code\app\client\node_modules\style-loader\addStyles.js

> WARNING in ./client/~/css-loader/lib/css-base.js
There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* D:\Code\app\client\node_modules\css-loader\lib\css-base.js
* d:\Code\app\client\node_modules\css-loader\lib\css-base.js

Any advice how to get rid of this?

**Comment by thorn0** [comment]
@whitecolor It turns out it makes a difference for Node on Windows whether you run `cd d:\foo` or `cd D:\foo` before running Node. So just use upper case everywhere in your paths. :trollface:

**Comment by whitecolor** [comment]
@thorn0 well but I don't set explicit name anywhere inside `webpack.config` I use `__dirname` and `path.resolve` and `__dirname` and `path.resolve` come with `D:/` where this small `d:` can appear from?

**Comment by thorn0** [comment]
@whitecolor As I wrote it comes from the properties of the shortcut or from the `cd` command.

**Comment by whitecolor** [comment]
I run `npm run webpack-dev` from command line. No any shortcut or cd commands.

I require some module from `node_modules` and it contains file that requires `.less` file inside this module, probably this cases to require `d:\Code\app\client\node_modules\css-loader\lib\css-base.js`. instead of 'D:/....'

**Comment by thorn0** [comment]
> No any shortcut or cd commands.

So how do you get to the command line then? How does your command prompt look? `d:\Code\app\client` or `D:\Code\app\client`?

**Comment by whitecolor** [comment]
`D:\Code\app\client`

**Comment by thorn0** [comment]
It even may be something inside LESS itself that changes the case.

**Comment by whitecolor** [comment]
Yes you are right. it is again about linked modules, that where linked using `d:` path. Thanks)

**Comment by jlroettger** [comment]
Another Windows user here, experienced the same issue. Again it was a drive letter that was the culprit--fixed it in `webpack.config.js` by changing my entry point from `path.join(__dirname, 'src')` (which uses a capital drive letter) to a hardcoded string `c:\\Users\\....\\src`

**Comment by b2whats** [comment]
may by we can disable this warning ?

**Comment by andrewmclagan** [comment]
In my case this was an `index.js` file actuallly names 'Index.js` and im on unix/linux

So be careful with
`import myModule from 'path/to/my/module/folder'`
rather then
`import myModule from 'path/to/my/module/folder/index.js'`

**Comment by mkontula** [comment]
Windows users, for me replacing `__dirname` with variable holding `__dirname.toLowerCase()` and using it consistently across webpack's config did the trick. No more warnings.

I'm using Cmder for everything console related and `pwd` command results in something like: `c:\node\my-project`. (lowercase C) However, no matter how I had `cd'd` in to that directory, `__dirname` was holding `"C:\node\my-project"` (UPPERCASE C) and I got plethora of warnings. 

It's a workaround, but I'm not quite sure if one should be forced to use hacks like this...

**Comment by jlroettger** [comment]
In the same vein as @mkontula , I am using this:

`path.join(__dirname, 'src').replace(/^([A-Z]:)/, v => v.toLowerCase())`

This prevents errors in Windows when `__dirname` contains capital letters elsewhere from the drive letter.

**Comment by alexananiev** [comment]
I have similar issue on Mac OS X, but in my case it seems like it starts happening after I've renamed my file (View.vue to view.vue). It looks like webpack thinks both file are exists, which isn't true.

```
WARNING in ./src/story/view.vue
There is another module with an equal name when case is ignored.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Rename module if multiple modules are expected or use equal casing if one module is expected.

WARNING in ./src/story/View.vue
There is another module with an equal name when case is ignored.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Rename module if multiple modules are expected or use equal casing if one module is expected.
```

**Comment by akfish** [comment]
I ran into this issue too. It behaves differently between PowerShell and cmd.exe.

Result of @thorn0 's `test.js` in PowerShell:
```
PS F:\dev\foo> node .\test.js
__dirname F:\dev\foo
__filename F:\dev\foo\test.js
require.resolve("./test") F:\dev\foo\test.js
require("path").resolve("test.js") F:\dev\foo\test.js
process.cwd() F:\dev\foo
```

In cmd.exe:
```
f:\Dev\foo>node test.js
__dirname f:\Dev\foo
__filename f:\Dev\foo\test.js
require.resolve("./test") f:\Dev\foo\test.js
require("path").resolve("test.js") f:\Dev\foo\test.js
process.cwd() f:\Dev\foo
```

**Comment by thorn0** [comment]
@AKFish What version of Node do you use?

On Aug 28, 2016 4:00 AM, "AKFish" <notifications@github.com> wrote:

I ran into this issue too. It behaves differently between PowerShell and
cmd.exe.

Result of @thorn0 <https://github.com/thorn0> 's test.js in PowerShell:

PS F:\dev\foo> node .\test.js
__dirname F:\dev\foo
__filename F:\dev\foo\test.js
require.resolve("./test") F:\dev\foo\test.js
require("path").resolve("test.js") F:\dev\foo\test.js
process.cwd() F:\dev\foo

In cmd.exe:

f:\Dev\foo>node test.js
__dirname f:\Dev\foo
__filename f:\Dev\foo\test.js
require.resolve("./test") f:\Dev\foo\test.js
require("path").resolve("test.js") f:\Dev\foo\test.js
process.cwd() f:\Dev\foo

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/webpack/webpack/issues/2362#issuecomment-242949537>,
or mute
the thread

**Comment by akfish** [comment]
@thorn0 I'm using node 6.4.0

PS version:
```
PS F:\dev> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.0.10586.494
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.10586.494
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by thorn0** [comment]
It has nothing to do with PowerShell itself. See my last comment in nodejs/node#6624.

**Comment by akfish** [comment]
@thorn0 

Try running `foo.bat`:
```
@echo off
echo %CD%
echo %~dp0
echo %0
```

In PowerShell:
```
"F:\dev\foo.bat"
PS F:\dev> .\foo.bat
F:\dev
F:\Dev\
"F:\dev\foo.bat"
```

In cmd.exe:
```
f:\Dev>foo.bat
f:\Dev
f:\Dev\
foo.bat
```

So it should have something to do with powershell. But I agree node.js should have normalized the drive letter.

**Comment by thorn0** [comment]
It has to do only with the way you start PowerShell and cmd (e.g. the 'Start in' property of the shortcut).

**Comment by akfish** [comment]
@thorn0 Not for powershell. It will normalize the drive letter to uppercase.

![image](https://cloud.githubusercontent.com/assets/922715/18054860/39836db0-6e38-11e6-8d4c-bc93556f8803.png)

**Comment by thorn0** [comment]
Yep, played a bit with PowerShell and was going to write the same.

**Comment by thorn0** [comment]
If PowerShell does so, it's a good argument for Node to do so as well.

**Comment by akfish** [comment]
@thorn0 Yep.

**Comment by jornare** [comment]
Using a capital letter in the drive of the path works for me. C:\ rather than c:\
Either change Your "start in" path, or from console:
cd C:\mypath     (the drive will not change)
d:                    (switch to another drive)
c:                   (switch back, and it's changed)

**Comment by MarkPieszak** [comment]
Just ran into this on one of my machines with Node 6.4, upgrading to 7.1 fixed the issue. Just a heads up for anyone who gets this far without fixing it :)

**Comment by Ciantic** [comment]
I'm struggling with this, I've used .replace("c:\\", "C:\\") in several place but still it doesn't work.

I think @thorn0's suggestion is right one:

> I think the best workaround here would be to make webpack internally consider D:\file.js and d:\file.js exactly same files.

This juggling with C: vs c: is maddening, the webpack should detect we are using Windows and deal with it.

**Comment by Ciantic** [comment]
Here is a emergency monkey patch:

E.g. top of your webpack.dev.server.js:

```
var WarnCaseSensitiveModulesPlugin = require('webpack/lib/WarnCaseSensitiveModulesPlugin');
WarnCaseSensitiveModulesPlugin.prototype.apply = function () {};
```

**Comment by 22ryann** [comment]
I fixed this by removing unnecessary capitol letters in my  'import {} from "HERE"' strings

In some cases i had written 
`import { Observable } from "rxjs/Rx"`
and
`import { Observable } from "rxjs/rx"`
fixed it.

I went through my project removing all caps where i did not need them and the errors went away. some camel case imports still just remain as is such as "mergeMap"

hope this helps someone.

Ryann,

**Comment by velobuff** [comment]
To beat a dead 🐴 

I had this issue on Mac OSX where I inadvertently imported a file from a location and capitalized the directory name i.e. `import ReactTableWrapper from '../../Components/ReactTableWrapper'`. It was happily imported however my directory name is `components`, not `Components`. 

After I realized this, I looked more carefully at the warnings which happily tell you there's a case difference i.e. `WARNING in ./src/containers/App/actions.js` `WARNING in ./src/Containers/App/actions.js` <-- see the difference? I fat fingered the folder name again and made it proper-case instead of lowercase. Now 💯  of those warnings are gone! :)

**Comment by UsulPro** [comment]
To get rid of problems with lowercase drive letter on Windows you can try this https://www.npmjs.com/package/driveup as a hotfix for this issue. It changes the drive letter the UpperCase:
```
e:\your_project_path>driveup
 
E:\your_project_path>
```
or to launch your shell:

```
e:\>startcmd cmder e:\your_project_path
```
You can setup cmder launching to always have uppercase drive letter

**Comment by szarouski** [comment]
I had that issue with NODE_PATH environment variable set to c:\... while path.join in webpack.config.js was using C:\... . Changed NODE_PATH to uppercase & problem went away.

**Comment by rajansen** [comment]
Or you can open a command window with shift-right mouse button => Open command window here .. from explorer

**Comment by cebence** [comment]
I was getting these warnings during `ng build` where the only difference is in the case of the drive letter (Windows 10):
```
WARNING in ./node_modules/@angular/platform-browser/esm5/platform-browser.js
There are multiple modules with names that only differ in casing.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Use equal casing. Compare these module identifiers:
* D:\my-project\node_modules\@angular\platform-browser\esm5\platform-browser.js
    Used by 9 module(s), i. e.
    D:\my-project\node_modules\@ngtools\webpack\src\index.js!D:\my-project\src\app\app.module.ts
* d:\my-project\node_modules\@angular\platform-browser\esm5\platform-browser.js
    Used by 2 module(s), i. e.
    d:\my-project\node_modules\@angular\forms\esm5\forms.js
```

This would mean that it has nothing to do with `import` statements containing capital letters (e.g. `@angular/Core`).
Also, I get warnings both in Command Prompt and PowerShell.
Even the way I switch directories made no difference: `cd /d d:\my-project` or `cd /d D:\my-project`.

Finally, I deleted the project's `node_modules` folder, and issued `npm install`, and `ng build` again -
 there were no warnings.

Could it have something to do with how NPM installs the dependencies, especially if dependencies were not added all in one go (i.e. days apart) and the drive letter case somehow changed in that time (e.g. `cd /d d:\my-project` and `cd /d D:\my-project`)?

**Comment by Ajaxy** [comment]
Removing `node_modules` and installing deps again helped me.

**Comment by alexander-akait** [comment]
We improve the errors in v5. Closing due to inactivity. Please test with latest version and feel free to reopen if you faced with this problem again or something will be misleading for you. Thanks!
